controllers/istio_controller.go

```diff
@@ -285,6 +285,10 @@ func (r *IstioReconciler) updateStatus(ctx context.Context, log logr.Logger, ist
 	}
 	status.AppliedValues = appliedValues
 
+	if reflect.DeepEqual(istio.Status, *status) {
```

Comment:
worth adding a simple comment reminding readers why this is necessary?

Reply:
It's pretty clear from the code itself. Why update the status if it hasn't changed? Additionally, the commit message fully explains why this is needed. I wouldn't want to add that long explanation here.